src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Feedback/RazorFeedbackDiagnosticFileProvider.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Globalization;
+using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
+using Microsoft.Internal.VisualStudio.Shell.Embeddable.Feedback;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Feedback
+{
+    [Shared]
+    [Export(typeof(IFeedbackDiagnosticFileProvider))]
+    internal class RazorFeedbackDiagnosticFileProvider : IFeedbackDiagnosticFileProvider
+    {
+        private readonly FeedbackLogDirectoryProvider _feedbackLogDirectoryProvider;
+        private readonly JoinableTaskFactory _joinableTaskFactory;
+
+        [ImportingConstructor]
+        public RazorFeedbackDiagnosticFileProvider(
+            JoinableTaskContext joinableTaskContext,
+            FeedbackLogDirectoryProvider feedbackLogDirectoryProvider)
+        {
+            if (joinableTaskContext is null)
+            {
+                throw new ArgumentNullException(nameof(joinableTaskContext));
+            }
+
+            if (feedbackLogDirectoryProvider is null)
+            {
+                throw new ArgumentNullException(nameof(feedbackLogDirectoryProvider));
+            }
+
+            _feedbackLogDirectoryProvider = feedbackLogDirectoryProvider;
+            _joinableTaskFactory = joinableTaskContext.Factory;
+        }
+
+        public IReadOnlyCollection<string> GetFiles()
+        {
+            if (!_feedbackLogDirectoryProvider.DirectoryCreated)
+            {
+                // No one requested to create any feedback logs, no reason for us to provide any logs.
+                return Array.Empty<string>();
+            }
+
+            var timestamp = DateTime.Now.ToString("yyyy-MM-dd-HH-mm-ss", CultureInfo.InvariantCulture);
+            var zipFileName = $"RazorLogs_{timestamp}.zip";
+            var zipFilePath = Path.Combine(Path.GetTempPath(), zipFileName);
+            var logDirectory = _feedbackLogDirectoryProvider.GetDirectory();
+            if (!Directory.Exists(logDirectory))
+            {
+                // This should never be the case, just being extra defensive.
+                return Array.Empty<string>();
+            }
+
+            _joinableTaskFactory.RunAsync(() => ZipLogsAsync(logDirectory, zipFilePath));
```

Comment:
Will this block? Or fire and forget?

Reply:
Fire and forget. This specific pattern is what he feedback team recommended so I pretty much just copy and pasted 😄